include/tvm/runtime/packed_func.h

```diff
@@ -1360,14 +1376,16 @@ inline bool TVMPODValue_::IsObjectRef() const {
   }
   // NOTE: we don't pass NDArray and runtime::Module as RValue ref.
   if (type_code_ == kTVMObjectRValueRefArg) {
-    return ObjectTypeChecker<TObjectRef>::Check(*static_cast<Object**>(value_.v_handle));
+    return !static_cast<bool>(
```

Comment:
IsObjectRef can be used in code-path that expects multiple argument variants. 

```
if (arg.IsObjectRef<T0>()) {
} else {
   another type of arg.
}
```

As a result, it is deseriiable to make the else path also fast(and not constructing error messages). Given that the recursive checking logic is not complicated. I think we could have separate impl for both CheckAndGetMismatchMessage and Check.